wgpu-core/src/device/mod.rs

```diff
@@ -321,6 +321,12 @@ impl<B: GfxBackend> Device<B> {
         };
 
         let mut buffer = unsafe { self.raw.create_buffer(desc.size, usage).unwrap() };
+        let label = conv::label_from_ptr(desc.label);
```

Comment:
I was just browsing the docs now, and noticed that `to_string_lossy` just borrows the existing contents when the string is a valid UTF8. I think that's all we really need here?
Do you think we could remove all the `Label` stuff and just call `to_string_lossy` on them before passing to gfx-hal?

Reply:
Or even just `to_str` and `log::warn` on being unable to get `Ok()` from it (ignoring the label in this case)

Reply:
> Do you think we could remove all the Label stuff and just call to_string_lossy on them before passing to gfx-hal?

This is what I had before.

Reply:
Deeply sorry if I missed that somehow! Could you link to that comment please?
I somehow assumed that all the conversion involve heap allocations before...

Reply:
https://github.com/gfx-rs/wgpu/pull/528/commits/fdcf9e7067c39147dba9944f97b9a1c896d35217

Maybe we should just hold off on this until gfx-hal is updated? The inlinable label stuff should be there anyway, as I described above.

Reply:
I'm not quite feeling comfortable with having `Label` in gfx-hal at this point. Maybe I need to grow to it? :)
Also, gfx-hal was just released, and we are updating wgpu to use it (#537). We typically try to stay on the published version of gfx, since otherwise it's harder to patch things up.

Let's proceed with your original code? (I slightly prefer `to_str` with warning on failure, but we can change that later)

Reply:
@kvark  I rolled off the last commit and force pushed. 

The inlinable is left here if there's any interest in it later: https://github.com/aloucks/wgpu/tree/buffer_label_with_inline_string